tensorbay/opendataset/BDD100K/loader.py

```diff
@@ -300,67 +314,146 @@ def _BDD100K_MOTS2020(path: str) -> Dataset:
-                _load_mots_data(
-                    image_subdir,
-                    os.path.join(labels_directory_segment, label_filename),
-                    segment,
-                )
-
-
-def _load_mots_data(image_subdir: str, label_path: str, target_segment: Segment) -> None:
-    with open(label_path, "r") as fp:
+            load_segment = _generate_data
+        for image_subdir in image_directory:
+            segment = dataset.create_segment(f"{segment_prefix}_{os.path.basename(image_subdir)}")
+            segment.extend(load_segment(image_subdir, labels_directory_segment, load_label))
+
+
+def _generate_test_data(image_subdir: str, _: str, __: _TYPE_LOAD_LABEL) -> Iterable[Data]:
+
+    yield from map(Data, _utility.glob(os.path.join(image_subdir, "*.jpg")))
+
+
+def _generate_data(
+    image_subdir: str,
+    labels_directory_segment: str,
+    load_label: _TYPE_LOAD_LABEL,
+) -> Iterable[Data]:
+    label_filename = f"{os.path.basename(image_subdir)}.json"
+    with open(os.path.join(labels_directory_segment, label_filename), "r") as fp:
         label_contents = json.load(fp)
     for label_content in label_contents:
         label_content_name = label_content["name"]
         if "/" in label_content_name:
             label_content_name = label_content_name[len(label_content["videoName"]) + 1 :]
         image_path = os.path.join(image_subdir, label_content_name)
 
-        data = Data(image_path)
-        labeled_multipolygons = []
-        for label_info in label_content.get("labels", ()):
-            if "poly2d" not in label_info:
-                continue
-            labeled_multipolygon = LabeledMultiPolygon(
-                polygons=(poly2d_info["vertices"] for poly2d_info in label_info["poly2d"]),
-                category=label_info["category"],
-                attributes=label_info["attributes"],
-                instance=str(label_info["id"]),
-            )
-            labeled_multipolygons.append(labeled_multipolygon)
-        data.label.multi_polygon = labeled_multipolygons
-
-        target_segment.append(data)
+        yield load_label(image_path, label_content)
+
+
+def _load_mot_label(image_path: str, label_content: Dict[str, Any]) -> Data:
+    data = Data(image_path)
+    labeled_box2ds = []
+    for label_info in label_content.get("labels", ()):
+        box2d_info = label_info.get("box2d", ())
+        if not box2d_info:
+            continue
+        labeled_box2d = LabeledBox2D(
+            box2d_info["x1"],
+            box2d_info["y1"],
+            box2d_info["x2"],
+            box2d_info["y2"],
+            category=label_info["category"],
+            attributes=label_info["attributes"],
+            instance=label_info["id"],
+        )
+        labeled_box2ds.append(labeled_box2d)
+    data.label.box2d = labeled_box2ds
+
+    return data
+
+
+def _load_mots_label(image_path: str, label_content: Dict[str, Any]) -> Data:
+    data = Data(image_path)
+    labeled_multipolygons = []
```

Comment:
`labeled_multipolygons` -> `multi_polygon`

Reply:
'labeled' should not be removed